android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/KeymanSettingsLocalizeActivity.java

```diff
@@ -0,0 +1,129 @@
+    getSupportActionBar().setDisplayShowHomeEnabled(true);
+    getSupportActionBar().setDisplayShowTitleEnabled(false);
+
+    final ListView listView = findViewById(com.tavultesoft.kmea.R.id.listView);
+
+    Bundle bundle = getIntent().getExtras();
+
+    final TextView textView = findViewById(R.id.bar_title);
+    textView.setText(getString(R.string.change_display_language));
+    if (titleFont != null) {
+      textView.setTypeface(titleFont, Typeface.BOLD);
+    }
+
+    localizeOptionList = new ArrayList<>();
+
+    final String noIcon = "0";
+    for(DisplayLanguages.DisplayLanguageType l: DisplayLanguages.DisplayLanguages) {
+      HashMap<String, String> hashMap = new HashMap<>();
+      hashMap.put(titleKey, l.getLanguageName());
+      String icon = String.valueOf(R.drawable.ic_content_add);
+      hashMap.put(iconKey, icon);
+      localizeOptionList.add(hashMap);
+    }
+
+    String[] from = new String[]{titleKey, subtitleKey, iconKey};
+    int[] to = new int[]{com.tavultesoft.kmea.R.id.text1, com.tavultesoft.kmea.R.id.text2, com.tavultesoft.kmea.R.id.image1};
+
+    ListAdapter adapter = new SimpleAdapter(context, localizeOptionList, com.tavultesoft.kmea.R.layout.list_row_layout2, from, to) {
+      /*
+      @Override
+      public boolean isEnabled(int position) {
+        HashMap<String, String> hashMap = localizeOptionList.get(position);
+        String itemTitle = MapCompat.getOrDefault(hashMap, titleKey, "");
+        String icon = MapCompat.getOrDefault(hashMap, iconKey, noIcon);
+        if (itemTitle.equals(getString(R.string.install_from_other_device))) {
+          // Scan QR code not implemented yet
+          return false;
+        } else if (itemTitle.equals(getString(R.string.add_languages_to_installed_keyboard))) {
+          if (KeyboardController.getInstance().getInstalledPackagesList() == null) {
+            // Disable if no keyboard packages installed
+            return false;
+          }
+        }
+
+        return super.isEnabled(position);
+      }
+       */
+    };
+    listView.setAdapter(adapter);
+    listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+
+      @Override
+      public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+        HashMap<String, String> hashMap = (HashMap<String, String>) parent.getItemAtPosition(position);
+
+        // Store the BCP-47 language tag in shared preference
+        if (position >= 0 && position < DisplayLanguages.DisplayLanguages.length) {
+
+          DisplayLanguages.DisplayLanguageType l = DisplayLanguages.DisplayLanguages[position];
+          String languageTag = l.getLanguageTag();
+
+          SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+          SharedPreferences.Editor editor = prefs.edit();
+          editor.putString(DisplayLanguages.displayLanguageKey, languageTag);
+          editor.commit();
+          finish();
```

Comment:
How does the user know they need to restart the app in order for the language to be reset? Is it possible to use [Activity.recreate](https://developer.android.com/reference/android/app/Activity.html#recreate%28%29) to avoid this user interaction?

Reply:
I've updated the activity to restart the app.